tests/lib/token-manager-test.js

```diff
@@ -630,7 +629,181 @@ describe('token-manager', function() {
 				.then(tokens => {
 					assert.equal(tokens, tokenInfo);
 				});
-		});
+		}).timeout(5000);
+
+		it('should retry with new jti claim when server rejects because of rate limiting', function() {
+
+			var firstTokenParams = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+				assertion: TEST_WEB_TOKEN
+			};
+
+			var newJWT = 'jhdhioeiu4yo34875twudgshdgr';
+			var secondTokenParams = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+				assertion: newJWT
+			};
+
+			var newJWT2 = 'jhighreiu4yo34875tfreggshdgr';
+			var thirdTokenParams = {
+				grant_type: 'urn:ietf:params:oauth:grant-type:jwt-bearer',
+				assertion: newJWT2
+			};
+
+			var serverDate = 'Sat, 01 Apr 2017 16:56:53 GMT'; // 1491065813
+
+			var serverError = {
+				statusCode: 429,
+				authExpired: false,
+				response: {
+					body: {
+						error: 'rate_limit_exceeded',
+						error_description: 'Request rate limit exceeded, please try again later'
+					},
+					headers: {
+						'retry-after': 1,
+						date: serverDate
+					}
+				}
+			};
+
+			var serverDate2 = 'Sat, 01 Apr 2017 16:56:58 GMT'; // 1491065818
+
+			var serverError2 = {
+				statusCode: 429,
+				authExpired: false,
+				response: {
+					body: {
+						error: 'rate_limit_exceeded',
+						error_description: 'Request rate limit exceeded, please try again later'
+					},
+					headers: {
+						'retry-after': 2,
+						date: serverDate2
+					}
+				}
+			};
+
+			var tokenInfo = {
+				accessToken: 'lsdjhgo87w3h4tbd87fg54'
```

Comment:
How was this generated? It should be a freshly-generated key from a non-Box account...

Reply:
This is taken from other tests in the file.